modules/pricer/src/main/java/com/opengamma/strata/pricer/swaption/SabrParametersSwaptionVolatilities.java

```diff
@@ -0,0 +1,630 @@
+
+import com.opengamma.strata.basics.PutCall;
+import com.opengamma.strata.basics.currency.Currency;
+import com.opengamma.strata.basics.date.DayCount;
+import com.opengamma.strata.basics.value.ValueDerivatives;
+import com.opengamma.strata.collect.ArgChecker;
+import com.opengamma.strata.collect.array.DoubleArray;
+import com.opengamma.strata.collect.tuple.DoublesPair;
+import com.opengamma.strata.market.sensitivity.SwaptionSabrSensitivity;
+import com.opengamma.strata.market.sensitivity.SwaptionSensitivity;
+import com.opengamma.strata.market.surface.NodalSurface;
+import com.opengamma.strata.market.surface.SurfaceCurrencyParameterSensitivities;
+import com.opengamma.strata.market.surface.SurfaceCurrencyParameterSensitivity;
+import com.opengamma.strata.market.surface.SurfaceMetadata;
+import com.opengamma.strata.market.surface.SurfaceParameterMetadata;
+import com.opengamma.strata.market.surface.meta.SwaptionSurfaceExpiryTenorNodeMetadata;
+import com.opengamma.strata.pricer.impl.option.BlackFormulaRepository;
+import com.opengamma.strata.pricer.impl.option.SabrInterestRateParameters;
+import com.opengamma.strata.product.swap.type.FixedIborSwapConvention;
+
+/**
+ * Volatility environment for swaptions in SABR model. 
+ * <p>
+ * The volatility is represented in terms of SABR model parameters.
+ * <p>
+ * The {@code surfaceCurrencyParameterSensitivity()}, {@code priceGamma()} and
+ * {@code priceTheta()} methods are not implemented.
+ */
+@BeanDefinition(builderScope = "private")
+public final class SabrParametersSwaptionVolatilities
+    implements SabrSwaptionVolatilities, ImmutableBean {
+
+  /** 
+   * The SABR model parameters. 
+   * <p>
+   * Each model parameter of SABR model is a surface in the expiry/swap tenor dimensions. 
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final SabrInterestRateParameters parameters;
+  /** 
+   * The swap convention. 
+   * <p>
+   * The data must valid in terms of this swap convention. 
+   */
+  @PropertyDefinition(validate = "notNull", overrideGet = true)
+  private final FixedIborSwapConvention convention;
+  /** 
+   * The valuation date-time. 
+   * <p>
+   * All data items in this environment are calibrated for this date-time. 
+   */
+  @PropertyDefinition(validate = "notNull", overrideGet = true)
+  private final ZonedDateTime valuationDateTime;
+  /** 
+   * The day count applicable to the model. 
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final DayCount dayCount;
+
+  //-------------------------------------------------------------------------
+  /**
+   * Obtains an instance from the SABR model parameters and the date-time for which it is valid.
+   * 
+   * @param parameters  the SABR model parameters
+   * @param convention  the swap convention for which the data is valid
+   * @param valuationDateTime  the valuation date-time
+   * @param dayCount  the day count applicable to the model
+   * @return the provider
```

Comment:
The return value isn't a provider, presumably that refers to an old version of the code.